core/store/statestore/state_batch.go

```diff
@@ -42,15 +43,24 @@ func NewStateStoreBatch(memoryStore common.MemoryCacheStore, store common.Persis
 
 func (self *StateBatch) Find(prefix common.DataEntryPrefix, key []byte) ([]*common.StateItem, error) {
 	var states []*common.StateItem
-	iter := self.store.NewIterator(append([]byte{byte(prefix)}, key...))
+	bp := []byte{byte(prefix)}
+	iter := self.store.NewIterator(append(bp, key...))
 	for iter.Next() {
 		key := iter.Key()
-		value := iter.Value()
-		state, err := getStateObject(prefix, value)
-		if err != nil {
-			return nil, err
+		keyV := key[1:]
+		if self.memoryStore.Get(byte(prefix), keyV) == nil {
+			value := iter.Value()
+			state, err := getStateObject(prefix, value)
+			if err != nil {
+				return nil, err
+			}
+			states = append(states, &common.StateItem{Key: string(keyV), Value: state})
+		}
+	}
+	for _, v := range self.memoryStore.Find() {
+		if v.State != common.Deleted && strings.HasPrefix(v.Key, string(append(bp, key...))) {
```

Comment:
define a prefix string `prefix := string(append(bp, key...))`  to avoid reconstruction in each iteration

Reply:
fix